src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -497,9 +499,27 @@ public override string MetadataName
         }
 
         /// <summary>
-        /// If this type is a file-local type, returns the syntax tree where this type is visible. Otherwise, returns null.
+        /// If this type is a file-local type, returns an identifier for the file this type was declared in. Otherwise, returns null.
         /// </summary>
-        internal abstract SyntaxTree? AssociatedSyntaxTree { get; }
+        internal abstract FileIdentifier? AssociatedFileIdentifier { get; }
+
+        internal struct FileIdentifier
+        {
+            public ImmutableArray<byte> FilePathChecksum { get; init; }
+            public string DisplayFilePath { get; init; }
+
+            public static FileIdentifier FromSyntaxTree(SyntaxTree tree)
```

Comment:
>FromSyntaxTree [](http://example.com/codeflow?start=41&length=14)

Consider moving this method out to the containing type, so callers do not need the `FileIdentifier` prefix.